src/System.Private.ServiceModel/src/System/ServiceModel/HttpBindingBase.cs

```diff
@@ -293,21 +269,11 @@ internal virtual void CheckSettings()
             {
                 return;
             }
-            else if (mode == BasicHttpSecurityMode.Message)
+            else if (mode == BasicHttpSecurityMode.Message || mode == BasicHttpSecurityMode.TransportWithMessageCredential)
             {
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException(SR.Format(SR.UnsupportedSecuritySetting, "Mode", mode)));
             }
 
-            // Message.ClientCredentialType = Certificate is not supported.
-            if (mode == BasicHttpSecurityMode.TransportWithMessageCredential)
-            {
-                BasicHttpMessageSecurity message = security.Message;
-                if ((message != null) && (message.ClientCredentialType == BasicHttpMessageCredentialType.Certificate))
-                {
-                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException(SR.Format(SR.UnsupportedSecuritySetting, "Message.ClientCredentialType", message.ClientCredentialType)));
-                }
-            }
```

Comment:
I'd rather this be a throw PlatformNotSupported where `BasicHttpSecurityMode.TransportWithMessageCredential`, if we're not supporting it rather than removing the code. 

The Assert a couple of lines down will still pass.